engine/src/test/java/io/camunda/zeebe/process/test/engine/EngineStateMonitorTest.java

```diff
@@ -10,40 +10,48 @@
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
+
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 class EngineStateMonitorTest {
 
+    private InMemoryLogStorage logStorage;
+    private TestLogStreamReader logStreamReader;
+    private EngineStateMonitor monitor;
+
+    @BeforeEach
+    void beforeEach() {
+        logStorage = mock(InMemoryLogStorage.class);
+        logStreamReader = new TestLogStreamReader();
+        monitor = new EngineStateMonitor(logStorage, logStreamReader);
+    }
+
   @Test
-  void testCallbackIsCalledWhenEngineIsAlreadyIdle()
+  void testOnIdleCallbackIsCalledWhenEngineIsAlreadyIdle()
       throws ExecutionException, InterruptedException, TimeoutException {
     // given
-    final InMemoryLogStorage logStorage = mock(InMemoryLogStorage.class);
-    final TestLogStreamReader logStreamReader = new TestLogStreamReader();
-    final EngineStateMonitor monitor = new EngineStateMonitor(logStorage, logStreamReader);
-
-    // when
     final CompletableFuture<Void> callbackFuture = new CompletableFuture<>();
     final Runnable callback = () -> callbackFuture.complete(null);
+
+    // when
+    changeToIdleState(monitor, logStreamReader, false);
     monitor.addOnIdleCallback(callback);
 
     // then
-    callbackFuture.get(50L, TimeUnit.MILLISECONDS);
+    callbackFuture.get(1L, TimeUnit.SECONDS);
     assertThat(callbackFuture).isCompleted();
   }
 
   @Test
-  void testCallbackIsCalledWhenIdleStateIsReached()
+  void testOnIdleCallbackIsCalledWhenIdleStateIsReached()
       throws ExecutionException, InterruptedException, TimeoutException {
     // given
-    final InMemoryLogStorage logStorage = mock(InMemoryLogStorage.class);
-    final TestLogStreamReader logStreamReader = new TestLogStreamReader();
-    final EngineStateMonitor monitor = new EngineStateMonitor(logStorage, logStreamReader);
-    changeToBusyState(monitor, logStreamReader);
-
-    // when
     final CompletableFuture<Void> callbackFuture = new CompletableFuture<>();
     final Runnable callback = () -> callbackFuture.complete(null);
+
+    // when
+    changeToBusyState(monitor, logStreamReader, false);
```

Comment:
It's not so clear to see which line in the when part is responsible for reaching the idle state.